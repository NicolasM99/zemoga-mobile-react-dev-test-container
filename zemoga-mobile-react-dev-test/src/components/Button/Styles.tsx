import { StyleSheet } from 'react-native';

const BORDER_WIDTH = 2;

export const buttonStyles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20
  },
  rounded: {
    borderRadius: 8
  },
  title: {
    color: 'white'
  },
  primary: {
    backgroundColor: 'blue'
  },
  secondary: {
    backgroundColor: 'green'
  },
  danger: { backgroundColor: 'red' },
  outlined_primary: {
    backgroundColor: 'white',
    borderWidth: BORDER_WIDTH,
    borderColor: 'blue'
  },
  outlined_secondary: {
    backgroundColor: 'white',
    borderWidth: BORDER_WIDTH,
    borderColor: 'green'
  },
  outlined_danger: {
    backgroundColor: 'white',
    borderWidth: BORDER_WIDTH,
    borderColor: 'red'
  },
  titleOutlined_primary: {
    color: 'blue'
  },
  titleOutlined_secondary: {
    color: 'green'
  },
  titleOutlined_danger: {
    color: 'red'
  }
});
