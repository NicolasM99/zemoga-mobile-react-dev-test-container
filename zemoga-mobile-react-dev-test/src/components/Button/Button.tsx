import React, { FC } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { IButton } from './@types/button';
import { buttonStyles } from './Styles';

const Button: FC<IButton> = ({
  title = '',
  variant = 'primary',
  style,
  outlined = false,
  rounded = true
}: IButton) => {
  return (
    <TouchableOpacity>
      <View
        style={{
          ...buttonStyles.container,
          ...buttonStyles[variant],
          ...(rounded && buttonStyles.rounded),
          ...style
        }}
      >
        <Text style={{ ...buttonStyles.title }}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Button;
