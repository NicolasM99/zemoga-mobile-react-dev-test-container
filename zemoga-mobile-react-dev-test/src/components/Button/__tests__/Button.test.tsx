import { render, screen } from '@testing-library/react-native';

import Button from '../Button';

describe('<Button /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<Button title="test" />);
    expect(screen).toMatchSnapshot();
  });
  it('Should render the button with correct label', async () => {
    render(<Button title="test" />);
    const buttonLabel = await screen.findByTestId('buttonLabel');
    expect(buttonLabel.props.children).toEqual('test');
  });
});
