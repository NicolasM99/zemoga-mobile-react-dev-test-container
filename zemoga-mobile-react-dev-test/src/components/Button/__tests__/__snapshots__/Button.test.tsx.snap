// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> unit tests Should mount correctly the component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#10c64e",
        "borderRadius": 8,
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
      testID="buttonLabel"
    >
      test
    </Text>
  </View>
</View>
`;
